assets/src/edit-story/components/canvas/useCanvasSelectionCopyPaste.js

```diff
@@ -108,41 +111,108 @@ function useCanvasSelectionCopyPaste(container) {
     [deleteSelectedElements, selectedElements]
   );
 
+  const elementPasteHandler = useCallback(
+    (html) => {
+      const template = document.createElement('template');
+      template.innerHTML = html;
+      let foundElements = false;
+      for (let n = template.content.firstChild; n; n = n.nextSibling) {
+        if (n.nodeType !== /* COMMENT */ 8) {
+          continue;
+        }
+        const payload = JSON.parse(
+          n.nodeValue.replace(new RegExp(DOUBLE_DASH_ESCAPE, 'g'), '--')
+        );
+        if (payload.sentinel !== 'story-elements') {
+          continue;
+        }
+        foundElements = true;
+        payload.items.forEach(({ x, y, basedOn, ...rest }) => {
+          currentPage.elements.forEach((element) => {
+            if (element.id === basedOn || element.basedOn === basedOn) {
+              x = Math.max(x, element.x + 60);
+              y = Math.max(y, element.y + 60);
+            }
+          });
+          const element = {
+            ...rest,
+            basedOn,
+            id: uuidv4(),
+            x,
+            y,
+          };
+          addElement({ element });
+        });
+      }
+      return foundElements;
+    },
+    [addElement, currentPage]
+  );
+
+  // @todo Split up this part to make the file smaller/more manageable.
+  const allowedContentNodes = ['strong', 'em', 'u'];
+
+  const processNodeContent = useCallback(
+    (node) => {
+      const tag = node.parentNode?.tagName?.toLowerCase();
+      const stripTags = !allowedContentNodes.includes(tag);
+      if (stripTags) {
+        return node.textContent.trim();
+      }
+      return `<${tag}>${node.textContent.trim()}<${tag}/>`;
+    },
+    [allowedContentNodes]
+  );
+
+  const processNodeList = useCallback(
+    (nodeList, content) => {
+      for (let i = 0; i < nodeList.length; i++) {
+        const n = nodeList[i];
+        if (n.childNodes.length > 0) {
+          if ('p' === n.tagName.toLowerCase() && content.trim().length) {
+            content += '\n';
+          }
+          content = processNodeList(n.childNodes, content);
+        } else {
+          content += processNodeContent(n, content);
+        }
+      }
+      return content;
+    },
+    [processNodeContent]
+  );
+
+  const rawPasteHandler = useCallback(
+    (html) => {
+      let foundContent = false;
+      const template = document.createElement('template');
```

Comment:
Could we put `const template = document.createElement(...); template.innerHTML = ...` to `pasteHandler` and use it to call `elementPasteHandler(tempate.content, ...)` and `rawPasteHandler(template.content, ...)`?